apis/python/src/tiledbsoma/io.py

```diff
@@ -189,66 +190,123 @@ def _from_anndata_aux(
     soma.create_unless_exists()
 
     # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-    soma.obs.from_dataframe(dataframe=anndata.obs, extent=256)
-    soma._add_object(soma.obs)
-
-    soma.var.from_dataframe(dataframe=anndata.var, extent=2048)
-    soma._add_object(soma.var)
+    max_workers = soma._soma_options.max_thread_pool_workers
 
     # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-    soma.X.add_layer_from_matrix_and_dim_values(
-        matrix=anndata.X,
-        row_names=anndata.obs.index,
-        col_names=anndata.var.index,
-        layer_name=X_layer_name,
-    )
-
-    if anndata.layers is not None:
-        for layer_name in anndata.layers:
-            soma.X.add_layer_from_matrix_and_dim_values(
-                matrix=anndata.layers[layer_name],
+    futures = []
+    with ThreadPoolExecutor(max_workers=max_workers) as executor:
+        futures.append(
```

Comment:
Any reason this isn't

```python
futures = [
    executor.submit(...),
    executor.submit(...),
    ...,
]
```
?

Reply:
Thanks for sharing your stylistic preference!